src/algos/fl_assigned.py

```diff
@@ -1,123 +1,111 @@
 import numpy as np
 import math
-
+from typing import Any, Dict, List
+from utils.communication.comm_utils import CommunicationManager
 from algos.base_class import BaseFedAvgClient, BaseFedAvgServer
 
 from utils.stats_utils import from_round_stats_per_round_per_client_to_dict_arrays
 
 
 class FedAssClient(BaseFedAvgClient):
-    def __init__(self, config) -> None:
-        super().__init__(config)
-
-    def get_collaborator_weights(self, round):
+    def __init__(self, config: Dict[str, Any], comm_utils: CommunicationManager) -> None:
+        super().__init__(config, comm_utils=comm_utils)
+        
```

Comment:
I will merge this time but for the future please do not leave trailing whitespaces